core/services/feeds/service.go

```diff
@@ -254,8 +258,45 @@ func (s *service) RegisterManager(ctx context.Context, params RegisterManagerPar
 	return id, nil
 }
 
-// SyncNodeInfo syncs the node's information with FMS
+// syncNodeInfoWithRetry syncs the node's information with FMS using a goroutine.
+// In case of failures, it retries with an exponential backoff for up to 24h.
+func (s *service) syncNodeInfoWithRetry(id int64) {
+	// cancel the previous context -- and, by extension, the existing goroutine --
+	// so that we can start anew
+	s.syncNodeInfoCancel()
+
+	var ctx context.Context
+	ctx, s.syncNodeInfoCancel = context.WithCancel(context.Background())
+
+	retryOpts := []retry.Option{
+		retry.Context(ctx),
+		retry.Delay(5 * time.Second),
+		retry.Delay(10 * time.Second),
+		retry.MaxDelay(30 * time.Minute),
+		retry.Attempts(48 + 8), // 30m * 48 =~ 24h; plus the initial 8 shorter retries
```

Comment:
Where did you derive the 8? where do we configure the shorter retries?

Reply:
I did a quick back of the envelope calculation based on the exponential backoff with a factor of 2, starting with 10s: (10s + 20s + 40s ... + 8th retry) =~ 30m.

Reply:
hmm, where do we configure the exponential backoff? All i see here is the fixed delay per retry, so the 48 + 8 attempts have 5 or 10 seconds delay between each attempt?

Reply:
What I understood (and saw in the tests) is that there's an exponential backoff that starts at `Delay` and double each time until we reach `MaxDelay`. So, we'd retry after:

1. 10s
2. 20s
3. 40s
4. 80s
5. 160s
6. 320s
7. 640s
8. 1280s
9. 1800s
10. 1800s
11. 1800s
...

> hmm, where do we configure the exponential backoff?

If you're asking about how to parameterize the delays, then I chose to leave it hardcoded. I didn't see the need to make it configurable (other than for tests; I'll need to think about this later).

Reply:
ah thanks! Got it, i misunderstood the library API, now i know it is exponential by default,